From cd7c8744a1b8ca152d4ac1f0829d0fe44d3e86a3 Mon Sep 17 00:00:00 2001
From: Mathieu Bivert <mathieu.bivert@intel.com>
Date: Tue, 30 Apr 2013 09:48:23 +0200
Subject: [PORT FROM MAIN][BT] crashtool: log fw version and loading failure

BZ: 104695 108245

Logs firmware loading failure.
Logs bluetooth fw version.

Change-Id: I42d4ab0738125d16aa0597429e3a9fb9f4f3ef80
Orig-Change-Id: Ice42706067f500de48c9a2cf8e86c19656f70a3d
Category: aosp improvement
Domain: CWS.BT-Common
Origin: internal
Upstream-Candidate: no, proprietary
Orig-Change-Id: If165312dae9baa4950aab10892a0926e3c183f2b
Signed-off-by: Mathieu Bivert <mathieu.bivert@intel.com>
Reviewed-by: Gros, Julien <julien.gros@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-on: http://android.intel.com:8080/123808
Reviewed-on: http://android.intel.com:8080/126132
Reviewed-by: Poignet, Cedric <cedric.poignet@intel.com>
---
 src/hardware.c |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/src/hardware.c b/src/hardware.c
index b609f3b..212789c 100644
--- a/src/hardware.c
+++ b/src/hardware.c
@@ -45,6 +45,8 @@
 #include "userial_vendor.h"
 #include "upio.h"
 
+#include <lct.h>
+
 /******************************************************************************
 **  Constants & Macros
 ******************************************************************************/
@@ -833,6 +835,7 @@ void hw_config_cback(void *p_mem)
                 {
                     uint16_t    lmp_subversion;
                     uint8_t     *p_lmp;
+                    char tmp[5];
 
                     p_lmp = (uint8_t *) (p_evt_buf + 1) + HCI_EVT_CMD_CMPL_LOCAL_REVISION;
                     STREAM_TO_UINT16(lmp_subversion, p_lmp);
@@ -842,6 +845,9 @@ void hw_config_cback(void *p_mem)
                         /* Found BCM4335B0 revision */
                         hw_cfg_cb.local_chip_name[7] = 'B';
                     }
+
+                    snprintf(tmp, sizeof(tmp), "%04x", lmp_subversion);
+                    lct_log(CT_EV_INFO, "cws.bt", "fw_version", 0, hw_cfg_cb.local_chip_name, tmp);
                 }
                 /* fall through intentionally */
 
@@ -860,6 +866,7 @@ check_local_name:
                         if ((hw_cfg_cb.fw_fd = open(tmp_path, O_RDONLY)) == -1)
                         {
                             ALOGE("vendor lib preload failed to open [%s]", tmp_path);
+                            lct_log(CT_EV_STAT, "cws.bt", "fw_error", 0, tmp_path);
                         }
                     }
                     else
@@ -867,6 +874,7 @@ check_local_name:
                         ALOGE( \
                         "vendor lib preload failed to locate firmware patch file" \
                         );
+                        lct_log(CT_EV_STAT, "cws.bt", "fw_error", 0, tmp_path);
                     }
                 }
 
@@ -1060,6 +1068,7 @@ check_local_name:
     if (is_proceeding == FALSE)
     {
         ALOGE("vendor lib fwcfg aborted!!!");
+        lct_log(CT_EV_STAT, "cws.bt", "fw_cfg", 0);
         if (bt_vendor_cbacks)
         {
             if (p_buf != NULL)
-- 
1.7.4.1

